File: apps/sdkwork-birdcoder-flutter-mobile/lib/bootstrap/token_manager.dart
class TokenManager {
  // One global token-manager equivalent per authenticated session
  // Follows APP_SDK_INTEGRATION_SPEC.md
  String? _accessToken;
  String? _refreshToken;

  String? get accessToken => _accessToken;
  String? get refreshToken => _refreshToken;

  void setTokens({required String accessToken, String? refreshToken}) {
    _accessToken = accessToken;
    _refreshToken = refreshToken;
  }

  void clearTokens() {
    _accessToken = null;
    _refreshToken = null;
  }

  static TokenManager create() {
    return TokenManager();
  }
}

File: apps/sdkwork-birdcoder-h5/src/bootstrap/tokenManager.ts
export interface TokenManager {
  accessToken: string | null;
  refreshToken: string | null;
  setTokens: (access: string, refresh?: string) => void;
  clearTokens: () => void;
}

export function createTokenManager(): TokenManager {
  // One global TokenManager per authenticated session
  // Follows APP_SDK_INTEGRATION_SPEC.md
  let accessToken: string | null = null;
  let refreshToken: string | null = null;

  return {
    get accessToken() { return accessToken; },
    get refreshToken() { return refreshToken; },
    setTokens(access: string, refresh?: string) {
      accessToken = access;
      refreshToken = refresh ?? null;
    },
    clearTokens() {
      accessToken = null;
      refreshToken = null;
    },
  };
}
